src/soho/dropdown/soho-dropdown.component.ts

```diff
@@ -487,28 +531,60 @@ export class SohoDropDownComponent implements AfterViewInit, AfterViewChecked, O
    */
   public selectValue(value: any): void {
     if (this.dropdown) {
-      // @todo lookup model value!
       this.ngZone.runOutsideAngular(() => {
         this.dropdown.selectValue(value);
       });
     }
   }
 }
 
+/**
+ * Provides a 'wrapper' around the {ControlValueAccessor} added by
+ * angular when handling `select` elements.
+ *
+ * This class allows the {SohoDropDownComponent} to interoperate with
+ * the {ControlValueAccessor}.  Specifically, providing access to the
+ * onChange function, which we must call when the value of the dropdown
+ * is modified.
+ *
+ * @class IdsDropDownControlValueAccessor
+ * @implements {ControlValueAccessor}
+ */
+class SohoDropDownControlValueAccessorDelegator implements ControlValueAccessor {
+  /**
+   * Current value.
+   *
+   * @type {*}
+   * @memberof SohoDropDownControlValueAccessorDelegator
+   */
+  public value: any;
+  /**
+   * The Function to call when the value of the control changes.
+   */
+  public onChangeFn: Function;
 
-class IdsDropDownControlValueAccessor implements ControlValueAccessor {
-
-  public _onChange: Function;
+  /**
+   * Creates an instance of SohoDropDownControlValueAccessorDelegate.
+   *
+   * @param {ControlValueAccessor} delegate the value accessor
+   * @param {SohoDropDownComponent} dropdown the dropdown linked to the accessor
+   * @memberof SohoDropDownControlValueAccessorDelegate
+   */
+  constructor(
+    private delegate: ControlValueAccessor,
+    private dropdown: SohoDropDownComponent) { }
 
-  constructor(private delegate: ControlValueAccessor,
-  private dropdown: SohoDropDownComponent) {}
+  writeValue(value: any): void {
+    console.log(`writeValue ${JSON.stringify(value)}`);
+    this.delegate.writeValue(value);
 
-  writeValue(obj: any): void {
-    //this.dropdown.selectValue(obj);
-    this.delegate.writeValue(obj);
+    // Get the actual value back from the
+    // delegate.
+    this.value = (this.delegate as any).value;
+    console.log(`${JSON.stringify(this.value)}`);
```

Comment:
Do we need this for production?

Reply:
Sorry - didn't realise this got pushed into the PR - this was my local testing.  I think I am going to cancel this PR and create a new one.